File: src/main/java/com/my/admin/app/commerce/service/validator/cash/CashFluxValidator.java
package com.my.admin.app.commerce.service.validator.cash;

import com.black.food.manager.model.api.request.cash.CreateCashRequest;
import com.black.food.manager.model.api.request.cash.UpdateCashRequest;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;

@Component
public class CashFluxValidator {

    private CashStatusValidator cashStatusValidator;

    public void validateRequestForCreate(final CreateCashRequest createCashRequest){
        cashStatusValidator.validateCashStatusRequest(createCashRequest.getStatusId());
    }

    public void validateRequestForUpdate(final UpdateCashRequest updateCashRequest){
        cashStatusValidator.validateCashStatusRequest(updateCashRequest.getStatusId());
    }

    @Autowired
    public void setCashStatusValidator(final CashStatusValidator cashStatusValidator) {
        this.cashStatusValidator = cashStatusValidator;
    }
}

File: src/main/kotlin/com/bk/reactive/app/commerce/admin/my/service/validator/cash/CashFluxValidator.kt
package com.bk.reactive.app.commerce.admin.my.service.validator.cash

import com.bk.reactive.app.commerce.admin.my.model.api.request.cash.CreateCashRequest
import com.bk.reactive.app.commerce.admin.my.model.api.request.cash.UpdateCashRequest
import org.springframework.stereotype.Component
import reactor.core.publisher.Mono

@Component
class CashFluxValidator(
    private val cashStatusValidator: CashStatusValidator
) {
    fun validateRequestForCreate(createCashRequest: CreateCashRequest): Mono<String> {
        return cashStatusValidator.validateCashStatusRequest(createCashRequest.statusId)
    }

    fun validateRequestForUpdate(updateCashRequest: UpdateCashRequest): Mono<String> {
        return cashStatusValidator.validateCashStatusRequest(updateCashRequest.statusId)
    }
}